typescript/sdk/src/deploy/HyperlaneDeployer.ts

```diff
@@ -76,11 +76,7 @@ export abstract class HyperlaneDeployer<
   ) {
     this.logger = options?.logger ?? debug('hyperlane:deployer');
     this.chainTimeoutMs = options?.chainTimeoutMs ?? 5 * 60 * 1000; // 5 minute timeout per chain
-
-    if (this.options?.ismFactory) {
-      this.options.ismFactory.deployContractFromFactory =
-        this.deployContractFromFactory.bind(this);
-    }
+    this.options.ismFactory?.setDeployer(this);
```

Comment:
@jmrossy @yorhodes I originally had the binding logic to avoid exposing `deployContractFromFactory`, however the logic seems to be a lot easier to understand this way